sdk/lib/opentelemetry/sdk/resources/detectors/telemetry.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module SDK
+    module Resources
+      module Detectors
+        # Telemetry contains detect class method for determining instrumentation resource labels
+        module Telemetry
+          extend self
+
+          def detect
```

Comment:
We will always have a Telemetry::SDK resource and we are required by the spec to create one. I'm not sure that this should be a detector, as I think auto detection should be optional.

Reply:
>We will always have a Telemetry::SDK resource and we are required by the spec to create one.

I missed that when going over the specs, I will remove this as a detector as you're right it does not belong in auto detection.   Could you link me to the spec that outline this (I haven't found it yet), I think adding this behaviour to the library fits this PR.